orc8r/cloud/go/obsidian/swagger/handlers/handlers.go

```diff
@@ -33,3 +100,79 @@ func GetGenerateCombinedSpecHandler(yamlCommon string) echo.HandlerFunc {
 		return c.String(http.StatusOK, combined)
 	}
 }
+
+// getSpecHandler returns a routing handler which creates and
+// serves the raw YAML spec of a particular service.
+func getSpecHandler(yamlCommon string) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		service, ok, err := getServiceName(c.Param("service"))
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+		if !ok {
+			return obsidian.HttpError(errors.New("service not found"), http.StatusNotFound)
+		}
+
+		combined, err := swagger.GetServiceSpec(yamlCommon, service)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+
+		return c.String(http.StatusOK, combined)
+	}
+}
+
+// getUIHandler returns a routing handler which serves
+// the UI of a service.
+func getUIHandler(tmpl *template.Template, enableDynamicSwaggerSpecs bool) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		requestedService := c.Param("service")
+
+		service, ok, err := getServiceName(requestedService)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+		if !ok {
+			return obsidian.HttpError(errors.New("service not found"), http.StatusNotFound)
+		}
+
+		// If runtime Swagger spec is off, serve static spec
```

Comment:
This code paragraph is wrong
- UI shouldn't care about dynamic swagger spec toggle
- If service is empty, just point to the `.../spec/` endpoint, which should serve the monolithic spec

The impl of the `.../spec/` endpoint is where we handle the feature toggle